spec/rubocop/cop/rspec/leaky_constant_declaration_spec.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::LeakyConstantDeclaration do
+  subject(:cop) { described_class.new }
+
+  describe 'constant assignment' do
+    it 'flags inside an example group' do
+      bad_code = <<-RUBY
+        describe SomeClass do
+          CONSTANT = 'is not allowed'.freeze
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Stub constant instead of declaring explicitly.
+        end
+      RUBY
+      expect_offense(bad_code)
+    end
+
+    it 'flags inside shared example group' do
+      bad_code = <<-RUBY
+        RSpec.shared_examples 'shared example' do
+          CONSTANT = 'is not allowed'.freeze
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Stub constant instead of declaring explicitly.
+        end
+      RUBY
+      expect_offense(bad_code)
+    end
+
+    it 'flags inside an example' do
+      bad_code = <<-RUBY
+        describe SomeClass do
+          specify do
+            CONSTANT = 'is not allowed'.freeze
+            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Stub constant instead of declaring explicitly.
+          end
+        end
+      RUBY
+      expect_offense(bad_code)
+    end
+
+    it 'ignores outside of example/shared group' do
+      fair_code = <<-RUBY
+        factory :some_class do
+          CONSTANT = 'is not allowed'.freeze
+        end
+      RUBY
+      expect_no_offenses(fair_code)
+    end
+  end
+
+  describe 'class defined' do
+    it 'flags inside an example group' do
+      bad_code = <<-RUBY
+        describe SomeClass do
+          class DummyClass < described_class
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Stub class constant instead of declaring explicitly.
+          end
+        end
+      RUBY
+      expect_offense(bad_code)
+    end
+
+    it 'ignores anonymous classes' do
```

Comment:
The anonymous class will also leak into global scope, won’t it?

Reply:
It will define a class, but that class won't clash by name with any other class.
The main problem this cop is addressing is that e.g. several examples may define a `DummyClass`, and instead of being a blank slate class as it will be in the first example, subsequent examples will be reopening it and modifying its behaviour in unpredictable ways. Another case when a class that exists in the codebase is redefined that way.
Do you think it makes sense to add this to cop's documentation?

Reply:
Yes, I think we should add a node that the cop allows anonymous classes. Should there be a bad/good example for it too?

Reply:
Added a note.

Existing good examples are using anonymous classes, so probably no additional example is needed.